src/type.ts

```diff
@@ -504,29 +504,46 @@ export class Type {
 
 export type Value = string | number | Date | boolean;
 export type ValueType = StringConstructor | NumberConstructor | DateConstructor | BooleanConstructor;
-export type EntityType = EntityConstructorForType<Entity>;
-export type PropertyType = ValueType | EntityType | ObjectConstructor;
+export type ReferenceType = EntityConstructorForType<Entity>;
```

Comment:
_info_: Rename `EntityType` to `ReferenceType` to avoid confusion with `EntityType` generic type arguments, and also is more consistent with reference/value terminology used in ExoModel.

Reply:
Might want to revert this...

Reply:
Reverted